keps/sig-api-machinery/4222-cbor-serializer/README.md

```diff
@@ -752,6 +753,42 @@ We expect no non-infra related flakes in the last month as a GA graduation crite
 
 - request and response content-type negotiation with 1.17 sample API server
 
+### Custom JSON Marshalers
+
+If a type implements json.Marshaler or json.Unmarshaler without corresponding CBOR behaviors,
+serializing values of that type to and from CBOR using default behaviors risks mangling the data.
+
+As an example, consider the structure of a marshalled
+[IntOrString](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/intstr#IntOrString) with the custom
+behavior versus the default behavior:
+
+| Go                                       | Custom  | Default                               |
+|------------------------------------------|---------|---------------------------------------|
+| IntOrString{Type: Int, IntVal: 7}        | 7       | {"IntVal":7,"StrVal":"","Type:":0}    |
+| IntOrString{Type: String, StrVal: "foo"} | "foo"   | {"IntVal":0,"StrVal":"foo","Type:":1} |
+| IntOrString{Type: -1}                    | <error> | {"IntVal":0,"StrVal":"","Type:":-1}   |
+
+Imagine a similar type is declared out-of-tree. From this type, a CRD and typed client are
+generated. This typed client is used in a program to write to a custom resource, using JSON to
+encode the request body. On the server side, the request body is decoded into an Unstructured
+object, and within that object, the IntOrString value is represented by either a `string` or an
+`int64`.
+
+Now imagine that the same request is repeated, but with CBOR as the negotiated content type of the
+request body. On the server side, the IntOrString value is represented within the decoded
+Unstructured as a `map[string]interface{}` with three keys, `"IntVal"`, `"StrVal`", and `"Type"`. A
+Change in the request encoding resulted in a change to the object the client intended to send.
```

Comment:
How would the server side handling a CRD know to use a map with these keys?  I would expect if the client sent either an int or a string as a value, that the server would handle it as unstructured (which doesn't care if it's an int or a string) and then CRD validation would check the value (if the server side were unaware of intOrString as a specially handled type, this validation would fail)

Reply:
I will clarify this section. Switching the client's preferred request encoding from JSON to CBOR means switching from the "custom" to the "default" encoding, unless this case is specially handled. So the JSON request body would contain a JSON number or JSON string, and the CBOR request body would contain a CBOR map.

Reply:
Updated, I hope that makes it clearer.